vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java

```diff
@@ -36,20 +46,31 @@
 
     @Autowired
     private IContentBlockManager contentBlockManager;
+    
+    @Autowired
+    private SpaceRepository spaceRepo;
 
     @RequestMapping("/staff/module/{moduleId}/slide/{id}")
-    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) {
+    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) throws JsonProcessingException {
 
         ISlide slide = slideManager.getSlide(id);
+        List<ISpace> spaces = new ArrayList<ISpace>();
+        spaceRepo.findAll().forEach(s -> {
+            spaces.add(s);
+        });
+        model.addAttribute("spaces", spaces);
         model.addAttribute("module", moduleManager.getModule(moduleId));
         model.addAttribute("slide", slide);
         model.addAttribute("slideSequences", slideManager.getSlideSequences(id, moduleId));
         List<IContentBlock> slideContents = contentBlockManager.getAllContentBlocks(id);
+        HashMap<String, String> selectedSpaceForSpaceBlock = contentBlockManager.findSelectedSpaceForSpaceBlock(slideContents);
         model.addAttribute("slideContents", slideContents);
+        model.addAttribute("selectedSpaceForSpaceBlock", selectedSpaceForSpaceBlock);
         model.addAttribute("contentCount",slideContents.size()>0 ? slideContents.get(slideContents.size()-1).getContentOrder() : 0);
         if(slideManager.getSlide(id) instanceof BranchingPoint) {
             model.addAttribute("choices", ((IBranchingPoint)slide).getChoices());
-        }           
+        }      
+        System.out.println("the map is"+ selectedSpaceForSpaceBlock.get("CON000000007"));
```

Comment:
Really? you even checked the box saying "I do not have any sysout statements in my code or commented out code that isn’t needed anymore" ;)